solidity/ecdsa/contracts/test/upgrades/WalletRegistryV2.sol

```diff
@@ -0,0 +1,1174 @@
+
+    event DkgParametersUpdated(
+        uint256 seedTimeout,
+        uint256 resultChallengePeriodLength,
+        uint256 resultSubmissionTimeout,
+        uint256 resultSubmitterPrecedencePeriodLength
+    );
+
+    event GasParametersUpdated(
+        uint256 dkgResultSubmissionGas,
+        uint256 dkgResultApprovalGasOffset,
+        uint256 notifyOperatorInactivityGasOffset,
+        uint256 notifySeedTimeoutGasOffset,
+        uint256 notifyDkgTimeoutNegativeGasOffset
+    );
+
+    event RandomBeaconUpgraded(address randomBeacon);
+
+    event WalletOwnerUpdated(address walletOwner);
+
+    event OperatorRegistered(
+        address indexed stakingProvider,
+        address indexed operator
+    );
+
+    event AuthorizationIncreased(
+        address indexed stakingProvider,
+        address indexed operator,
+        uint96 fromAmount,
+        uint96 toAmount
+    );
+
+    event AuthorizationDecreaseRequested(
+        address indexed stakingProvider,
+        address indexed operator,
+        uint96 fromAmount,
+        uint96 toAmount,
+        uint64 decreasingAt
+    );
+
+    event AuthorizationDecreaseApproved(address indexed stakingProvider);
+
+    event InvoluntaryAuthorizationDecreaseFailed(
+        address indexed stakingProvider,
+        address indexed operator,
+        uint96 fromAmount,
+        uint96 toAmount
+    );
+
+    event OperatorJoinedSortitionPool(
+        address indexed stakingProvider,
+        address indexed operator
+    );
+
+    event OperatorStatusUpdated(
+        address indexed stakingProvider,
+        address indexed operator
+    );
+
+    event InactivityClaimed(
+        bytes32 indexed walletID,
+        uint256 nonce,
+        address notifier
+    );
+
+    modifier onlyStakingContract() {
+        require(
+            msg.sender == address(staking),
+            "Caller is not the staking contract"
+        );
+        _;
+    }
+
+    /// @notice Reverts if called not by the Wallet Owner.
+    modifier onlyWalletOwner() {
+        require(
+            msg.sender == address(walletOwner),
+            "Caller is not the Wallet Owner"
+        );
+        _;
+    }
+
+    modifier onlyReimbursableAdmin() override {
+        require(governance == msg.sender, "Caller is not the governance");
+        _;
+    }
+
+    /// @dev Used to initialize immutable variables only, use `initialize` function
+    ///      for upgradable contract initialization on deployment.
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(SortitionPool _sortitionPool, IStaking _staking) {
+        sortitionPool = _sortitionPool;
+        staking = _staking;
+    }
+
+    /// @dev Initializes upgradable contract on deployment.
+    function initialize(
+        DKGValidator _ecdsaDkgValidator,
+        IRandomBeacon _randomBeacon,
+        ReimbursementPool _reimbursementPool
+    ) external initializer {
+        // TEST: Removed initializer of V1
```

Comment:
We do not need to remove it, right? The `initializer` should protect against calling this function another time. Are we trying to get some bytecode back this way? Could be worth dropping a comment on why it is removed.

Reply:
Yeah, I think we could even remove the `initialize` function completely.

Reply:
b43df059dbd52682c5e8ce8f42467afb2be3e358